tests/flytekit/unit/extras/sqlalchemy/test_task.py

```diff
@@ -0,0 +1,98 @@
+import os
+import pandas
+import pytest
+import shutil
+from subprocess import Popen
+import tempfile
+import time
+
+import doltcli as dolt
+
+from flytekit import current_context, kwtypes, Secret, task, workflow
+from flytekit.extras.sqlalchemy.task import SQLAlchemyConfig, SQLAlchemyTask
+
+# https://www.sqlitetutorial.net/sqlite-sample-database/
+from flytekit.types.schema import FlyteSchema
+
+OK_EXAMPLE_DB = "mysql+pymysql://root@localhost:3307/tracks"
+BAD_EXAMPLE_DB = "mysql+pymysql://root1@localhost/tracks"
+
+# This task belongs to test_task_static but is intentionally here to help test tracking
+tk = SQLAlchemyTask(
+    "test",
+    query_template="select * from tracks",
+    task_config=SQLAlchemyConfig(
+        uri=OK_EXAMPLE_DB,
+    ),
+)
+
+@pytest.fixture(scope="function")
+def sql_server():
+    p = None
+    try:
+        d = tempfile.TemporaryDirectory()
+        db_path = os.path.join(d.name, "tracks")
+        db = dolt.Dolt.init(db_path)
```

Comment:
interesting that dolt is only a test dependencyy

Reply:
this integration will work with any sqlalchemy dialect. Dolt is probably not the best way to test this, honestly. Setting up docker-compose resources for mysql, postgres, and a few others to test the integration is a lot of code, though https://haseebmajid.dev/blog/gitlab-ci-pytest-and-docker-compose. Other alternatives work, that's how i'd do it though.